### DISCUSSION TITLE
Screen update based on Entity State

### DISCUSSION THREAD

**Discussion by davidlamd89** [pixelit-project/PixelIt#234] [Q&A]
I need some help, I would like to know if there is a simple way to update the playlist based on the state of an entity and when stopped or off remove the screen from rotation.  
I'm using 5 screens, one that's always on rotation that displays temperature, 3 media players showing what is currently playing and one for my printer to represent the percentage during printing, but the logic becomes complicated.  
It's my first time using Node-Red and I'm probably making this too complicated.

Thanks in advance!

![IMG_20221116_151800.jpg](https://user-images.githubusercontent.com/29853028/202316564-cc706358-2c80-49b4-8a23-9cae473b7c92.jpg)

**Comment by hamster65** [comment]
Looks nice, but probably, next time. use a screenshot :)
You can set msg.show = false to disable or enable screens.  Here is an example of a function do do this based on a value. https://github.com/pixelit-project/PixelIt/discussions/231

**Reply by davidlamd89** [reply]
You're right, I should have taken a screenshot, but I just wanted to demonstrate the mess I was creating. I will see the example.

Thank you for your help.

**Comment by davidlamd89** [comment]
The example works great and my config is now much simpler. 

![Node Red - PixelIT](https://user-images.githubusercontent.com/29853028/202556302-19f12c77-4576-41a2-b01c-4e445a8c328a.PNG)

I was wondering if it's possible to add the media info sensor (Emby/Spotify etc.) to the function that controls the display and make it even simpler.

![image](https://user-images.githubusercontent.com/29853028/202557253-71ff9ea2-65b3-4389-9d8e-291ab708db0a.png)

**Reply by hamster65** [reply]
Great! What's the node with the "i" in the circle?

**Reply by davidlamd89** [reply]
It's a template sensor that grabs the name of the movie or tv show that Emby is playing. I would like to add that info to the function node, but I don't know how.

**Comment by davosje** [comment]
Hi @davidlamd89, I'm very interested in your configuration, are you willing to share (pieces)? The mediaplayer and octoprint part are the ones looking for. Just started using node-red, thanks in advance 😄

**Comment by aletdf** [comment]
Hi @davidlamd89, I also want to show what is playing in my android tv, what is the function you are using to show media?
can you share your json?

**Reply by aletdf** [reply]
Ok, I found the node and the variable. With "current state" I call my TV. And the info on what is playing is in msg.data.attributes.media_title
You can also get app running, time, and almost everything from the media.

**Comment by Waynee91** [comment]
Hey @davidlamd89, can you may share the code from your node?
You have functions I also want to implement such as showing the temperature.
I'm quite at the beginning and a total newby in NodeRed :D 
Would be appreciated, thanks!
